internal/mode/static/state/graph/backend_refs.go

```diff
@@ -142,16 +149,33 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
-	svcNsName, svcPort, err := getServiceAndPortFromRef(ref.BackendRef, sourceNamespace, services, refPath)
+	ns := sourceNamespace
```

Comment:
Sorry, I know we chatted about this on slack but seeing the code I think we can improve this a bit. 

Let's pass the svcNsName into `getServiceAndPortFromRef` and also rename that function to reflect it's new purpose which is to get the IPFamily and port for the ref:

```
svcIPFamily, svcPort, err := getIPFamilyAndPortFromRef(ref.BackendRef, svcNsName, services, refPath)
```